routers/user/setting/adopt.go

```diff
@@ -30,10 +30,17 @@ func AdoptOrDeleteRepository(ctx *context.Context) {
 	root := filepath.Join(models.UserPath(ctxUser.LowerName))
 
 	// check not a repo
-	if has, err := models.IsRepositoryExist(ctxUser, dir); err != nil {
+	has, err := models.IsRepositoryExist(ctxUser, dir)
+	if err != nil {
 		ctx.ServerError("IsRepositoryExist", err)
 		return
-	} else if has || !com.IsDir(filepath.Join(root, dir+".git")) {
+	}
+
+	isDir, err := util.IsDir(filepath.Join(root, dir+".git"))
+	if err != nil {
+		ctx.ServerError("IsDir", err)
+	}
```

Comment:
```return``` should be here, you can not continue on error

Reply:
Thanks, it was a rather tiresome PR and I must have just forgotten to add these.